doc/man3/OSSL_trace_enabled.pod

```diff
@@ -136,26 +152,71 @@ This will normally expand to:
  } while (0);
 
 
-C<OSSL_TRACE1()>, ... C<OSSL_TRACE9()> are one-shot macros which essentially wrap
-a single BIO_printf() into a tracing group.
+C<OSSL_TRACE()> and C<OSSL_TRACE1()>, C<OSSL_TRACE1()>, ... C<OSSL_TRACE9()> are
+so-called one-shot macros:
+
+The macro call C<OSSL_TRACE(category, text)>, produces literal text trace output.
+
+The macro call C<OSSL_TRACEn(category, format, ...arg1, ..., argn)> produces
+printf-style trace output with n format field arguments (n=1,...9).
+It expands to:
+
+ OSSL_TRACE_BEGIN(category) {
+     BIO_printf(trc_out, format, arg1, ..., argN)
+ } OSSL_TRACE_END(category)
+
+Internally, all one-shot macros are implemented using a generic C<OSSL_TRACEV()>
+macro, since C90 does not support variadic macros. This helper macro has a rather
+weird synopsis and is not intended to be used directly.
```

Comment:
I'd make this stronger. Perhaps "must not be used directly".

Reply:
Hmmm... "must" sounds a bit too strong IMHO. Can we meet in the middle and say "_should_ not be used directly"?

On one hand, I don't want to encourage using `OSSL_TRACEV()` by documenting it (mainly because of its odd synopsis), but on the other hand, I don't want to forbid it either. One valid use case for an application developer would be for example to add more macros `OSSL_TRACE10(), ..., OSSL_TRACE11()` along the lines of the existing ones.

_(So it's more like "Use it at your own risk. We're unlikely to change it, but we don't advertise it as an official api.")_

Reply:
Its the last bit I was mostly worried about, i.e. people using it and expecting it to be stable.  I'd be ok with "should", if there was something about it possibly changing in the future.

Reply:
This helper macro is really unlikely to change, since the 'numbered' macros derived from it are officially documented and not allowed to change after their initial release. (Currently, the new TRACE api only exists on master.)

Reply:
I strongly disagree with "must". These are public macros, we cannot dictate their use, only give the users a "fair warning". Our users often use more modern compilers.

Reply:
Ok, fair enough.

Reply:
I like how fair we are :wink: